### PR TITLE
Fixing direct method calls (not through the handler) writing messages in terminal

### PR DESCRIPTION
**Pull request by sosadchuk**
Added environment variables support (`ZOWE_APP_LOG_LEVEL`, `ZOWE_IMPERATIVE_LOG_LEVEL`) for setting console logging level in order to fix [this issue](https://github.com/zowe/zowe-cli/issues/511).

Short description: we're getting logger messages in terminal upon calling zowe methods directly in test code:
```
[2022/06/08 13:25:50.300] [INFO] [Logger.ts:201] The current status (OUTPUT) for jobname EXECSQL, jobid JOB53810 is greater than or equal to the expected of "OUTPUT"

[2022/06/08 13:25:50.301] [INFO] [Logger.ts:201] Expected status "OUTPUT" found for jobname "EXECSQL" jobid "JOB53810" - attempt "3" (max attempts "Infinity") ...

[2022/06/08 13:25:50.302] [INFO] [Logger.ts:183] Monitor Jobs - "waitForStatusCommon" complete - found expected status of OUTPUT

[2022/06/08 13:25:50.302] [INFO] [Logger.ts:183] GetJobs.getSpoolFilesForJob()

[2022/06/08 13:25:50.303] [INFO] [Logger.ts:183] GetJobs.getSpoolFilesCommon()

[2022/06/08 13:25:50.306] [INFO] [Logger.ts:219] GetJobs.getSpoolFilesCommon() parameters: /EXECSQL/JOB53810/files
```

Signed-off-by: Sergei Osadchuk <sergei.osadchuk@broadcom.com>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 All notable changes to the Imperative package will be documented in this file.
 
+## Recent changes
```
Comment (sosadchuk) [inline CHANGELOG.md:5]: Sorry, this should have been `## Recent Changes` with capitalized second word, will push additional commit in a moment.

**Comment by sosadchuk** [comment]
Hi @t1m0thyj Timothy, I think there is some issue with passing `SONAR_TOKEN` to the `SonarCloud Scan` step. 

You might enable using secrets for fork PRs:
image

Alternatively you could switch from using the `pull_request` to the `pull_request_target` to avoid any issues as described here:
https://github.blog/2020-08-03-github-actions-improvements-for-fork-and-pull-request-workflows/

Thanks,
Sergei.

**Comment by t1m0thyj** [comment]
@sosadchuk Thanks for your contribution 🙂 

I've discussed this PR with the team and we have some concerns about how it may affect default behavior for other apps besides Zowe CLI that use the Imperative SDK to handle logging. We are tied up with other commitments this week, but will discuss this further and get back to you when possible.

Regarding the SonarCloud scan, we don't want to enable secrets for fork PRs for security reasons. We're in the process of migrating our CI from Jenkins to GitHub Actions which is probably why the step broke, so we'll look into fixing it.

**Comment by sosadchuk** [comment]
> @sosadchuk Thanks for your contribution 🙂
> 
> I've discussed this PR with the team and we have some concerns about how it may affect default behavior for other apps besides Zowe CLI that use the Imperative SDK to handle logging. We are tied up with other commitments this week, but will discuss this further and get back to you when possible.
> 
> Regarding the SonarCloud scan, we don't want to enable secrets for fork PRs for security reasons. We're in the process of migrating our CI from Jenkins to GitHub Actions which is probably why the step broke, so we'll look into fixing it.

Hi Timothy, any updates on this one or any changes needed from my side?

**Comment by t1m0thyj** [comment]
> Hi Timothy, any updates on this one or any changes needed from my side?

Hi @sosadchuk, discussed this with the team and we think there is a bug here:
https://github.com/zowe/imperative/blob/2ace4a586a7d211ac91cc5b2fb0ee7fed993fc44/packages/console/src/Console.ts#L26

In Zowe v2 we changed the default log level for Zowe CLI from DEBUG to WARN, but missed the change in the Console API so its default level is still DEBUG.

Would changing the default log level of the Console API from DEBUG to WARN fix the problem you have experienced with Imperative logging being too verbose?

**Comment by t1m0thyj** [comment]
Hi @sosadchuk, we have changed the default log level of the Console API from DEBUG to WARN in Imperative v5.4.1.

Could you please try updating to this version of Imperative, and let us know if it fixes your issue of too many messages being written to the terminal?

**Comment by sosadchuk** [comment]
> Hi @sosadchuk, we have changed the default log level of the Console API from DEBUG to WARN in Imperative v5.4.1.
> 
> Could you please try updating to this version of Imperative, and let us know if it fixes your issue of too many messages being written to the terminal?

Hi @t1m0thyj , messages are still there. I think the reason behind that might be that @zowe/cli package published still relies on @zowe/imperative@5.3.8, will try to apply overrides for it.

**Comment by t1m0thyj** [comment]
> Hi @t1m0thyj , messages are still there. I think the reason behind that might be that @zowe/cli package published still relies on @zowe/imperative@5.3.8, will try to apply overrides for it.

If applying overrides is inconvenient, we can wait to verify the fix until Zowe CLI has been updated to use the new version of Imperative 🙂

**Comment by sosadchuk** [comment]
> > Hi @t1m0thyj , messages are still there. I think the reason behind that might be that @zowe/cli package published still relies on @zowe/imperative@5.3.8, will try to apply overrides for it.
> 
> If applying overrides is inconvenient, we can wait to verify the fix until Zowe CLI has been updated to use the new version of Imperative 🙂

Yeah, would be nice to double-check. Thx!

**Comment by t1m0thyj** [comment]
@sosadchuk Could you please try Zowe CLI v7.5.0? It has been updated to use the new version of Imperative.

**Comment by sosadchuk** [comment]
> @sosadchuk Could you please try Zowe CLI v7.5.0? It has been updated to use the new version of Imperative.

Hi @t1m0thyj , it looks great now (meaning we don't see redundant log output), thank you! :)

**Pull request closed without merging**